src/storage/core.rs

```diff
@@ -897,30 +901,33 @@ where
     async fn mark_all_as_deleted_closed(&self, key: &K) -> Result<u64> {
         let safe = self.inner.safe.write().await;
         let mut blobs = safe.blobs.write().await;
-        let entries_closed_blobs = blobs
-            .iter_mut()
-            .map(|b| b.mark_all_as_deleted(key))
-            .collect::<FuturesUnordered<_>>();
-        let total = entries_closed_blobs
-            .filter_map(|result| match result {
-                Ok(count) => count,
-                Err(error) => {
-                    warn!("failed to delete records: {}", error);
-                    None
+        let count = AtomicU64::new(0);
+        blobs
+            .for_each(|b| async {
+                let rwlock = ASRwLock::new(b);
+                let result = Blob::mark_all_as_deleted(&rwlock, key).await;
+                match result {
+                    Ok(cnt) => {
+                        count.fetch_add(cnt.unwrap_or(0), Ordering::Release);
+                    }
+                    Err(err) => warn!("failed to delete records: {}", err),
```

Comment:
Can you check why this error is not propagated above?

Reply:
It would be good to return error to the Bob client if the error happened somewhere inside. But we should not stop processing other blobs if error happened in one

Reply:
Yep, error is returned if any blobs contained error, but processing is not stopped